sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ReceiveMessageAndSettleAsyncSample.java

```diff
@@ -45,18 +45,22 @@ public static void main(String[] args) {
             .buildAsyncClient();
 
         Disposable subscription = receiverAsyncClient.receive()
-            .map(message -> {
+            .flatMap(message -> {
```

Comment:
why flatmap?

Reply:
Because of the `map`  will not propagated subscribe to internal Mono.  Thus internal .complete() call was not getting invoked.